generic/piranha-tree-sitter/src/cleanup_rules/java/rules.toml

```diff
@@ -0,0 +1,540 @@
+                    right: (_) @a.rhs) @a)
+(#eq? @a.lhs "@variable_name")
+(#not-eq? @a.rhs "@init")
+)"""]
+
+
+# This rule is part (and entry point) of the inline field declaration cleanup.
+# Same as the rule `delete_variable_declaration`, but applicable to 
+# fields. 
+[[rules]]
+name = "delete_field_declaration"
+query = """
+(
+((field_declaration 
+				declarator: (variable_declarator 
+                					name: (_) @variable_name
+                					value: [(true) (false) (parenthesized_expression (true)) (parenthesized_expression (false))] @init)) @field_declaration)
+)
+"""
+replace_node = "field_declaration"
+replace = ""
+groups = []
+# Check if there is no assignment where the variable @variable_name is 
+# assigned to a value other than @init, within the method body
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = ["""(
+((assignment_expression
+    				left: (_) @a.lhs
+                    right: (_) @a.rhs) @a)
+(#eq? @a.lhs "@variable_name")
+(#not-eq? @a.rhs "@init")
+)"""]
+
+
+# This rule is part of inline variable declaration and field declaration cleanup. 
+# This rule is applied after the above two rule - `delete_field_declaration` and `delete_variable_declaration`.
+# Delete assignment for a particular (@variable_name, @init) if: 
+# (i) If the enclosing method does not contain a variable declaration for @variable_name
+# (ii) If the Rhs of the assignment is same as the initializer of the variable it previously inlined.
+[[rules]]
+name = "delete_boolean_assignments_for_variable"
+query = """
+(
+(expression_statement (assignment_expression
+                                left: (_) @l
+                    right: (_) @r)) @expression_statement
+(#eq? @l "@variable_name")
+(#eq? @r "@init")
+) 
+"""
+replace_node = "expression_statement"
+replace = ""
+groups = []
+holes = ["variable_name", "init"]
+[[rules.constraints]]
+matcher = "(method_declaration) @md"
+queries = ["""(
+((local_variable_declaration 
+				declarator: (variable_declarator 
+                					name: (_) @vdcl.lhs
+                					value: (_) @vdcl.init)) @field_declaration)
+(#eq? @vdcl.lhs "@l")
+)"""]
+
+# This rule is part (and entry point) of the inline variable / field declaration. 
+# After applying this rule, the two rules `delete_field_declaration` and `delete_variable_declaration` are applied. 
+# delete assignment if 
+# (i) if it is the only assignment in the entire class in the enclosing class
+# (ii) There is a field declaration with the same rhs as this assignment in the enclosing class
+# (iii) There is no local variable declaration of the same name in the enclosing method
+[[rules]]
+name = "delete_parent_assignment"
+query = """
+(
+(expression_statement (assignment_expression
+                                left: (_) @variable_name
+                    right: (_) @init)) @expression_statement
+) 
+"""
+replace_node = "expression_statement"
+replace = ""
+groups = []
+holes = []
```

Comment:
Why are these holes empty when the holes for `delete_boolean_assignments_for_variable` are non-empty?

Reply:
yea i ll remove wherever field is initialized to `[]`